client/app.jsx

```diff
@@ -40,27 +75,36 @@ export default class App extends React.Component {
 
   handleSearch(result) {
     const games = result;
-    this.setState({ listOfGames: games });
+    this.setState({ listOfGames: games, currentPage: 1 });
   }
 
   renderPage() {
+    const indexOfLastPost = this.state.currentPage * this.state.postsPerPage;
+    const indexOfFirstPost = indexOfLastPost - this.state.postsPerPage;
     const { route } = this.state;
     const searchTerm = route.params.get('term');
     const gameId = route.params.get('gameId');
     const { listOfGames } = this.state;
+    const currentListOfGames = listOfGames === null
+      ? null
```

Comment:
You have a few of these `null` checks, please try reworking your code so that they are not necessary. It's safe to slice empty arrays, just use empty arrays where there are no games or comments.